libcxx/include/__string/constexpr_c_functions.h

```diff
@@ -35,18 +35,23 @@ _LIBCPP_BEGIN_NAMESPACE_STD
 // of elements as opposed to a number of bytes.
 enum class __element_count : size_t {};
 
-inline _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX14 size_t __constexpr_strlen(const char* __str) {
+template <class _Tp>
```

Comment:
I really don't like that we're accepting arbitrary integral types in `__constexpr_strlen`. IMO it doesn't make sense to use `strlen` on e.g. `int8_t` -- it only makes sense for character types. Instead, I'd be much happier if you checked explicitly for `char` and `char8_t`.